lib/conversion.c

```diff
@@ -123,17 +124,22 @@ static bool tpm2_convert_pubkey_ssl(TPMT_PUBLIC *public, pubkey_format format, c
         goto error;
     }
 
-    ssl_rsa_key->e = BN_bin2bn((void*)&exponent, sizeof(exponent), NULL);
-    ssl_rsa_key->n = BN_bin2bn(
-            (public->unique).rsa.t.buffer,
-            (public->unique).rsa.t.size,
-            NULL);
+    e = BN_bin2bn((void*)&exponent, sizeof(exponent), NULL);
+    n = BN_bin2bn(public->unique.rsa.t.buffer, public->unique.rsa.t.size,
+                  NULL);
 
-    if (!ssl_rsa_key->n || !ssl_rsa_key->e) {
+    if (!n || !e) {
```

Comment:
When only one of these conversions fails then we'll leak the other one, because no `BN_free` is called on it.

This is tricky, because once the BIGNUMs are successfully stored in the `ssl_rsa_key` structure, `RSA_free` will take care of freeing that memory.

Maybe tracking the ownership of `e` and `n` can be done with an extra `bool` variable that triggers the appropriate `BN_free` in the error handling code at the end of the function.

Reply:
Maybe just NULL out n and e when the ownership transfer occurs.

Reply:
@mgerstner thanks a lot for pointing out these issues. I'm not really familiar with the OpenSSL lAPI so I missed these details.

@williamcroberts agreed that the cleaner option is to NULL these when are owned by the RSA struct. I'll re-spin the commit.

Reply:
@mgerstner @williamcroberts does it look Ok for you now?

Reply:
It looks good to me, thank you.